test/tools/router_check/test/route_tests.sh

```diff
@@ -16,6 +16,19 @@ do
   TEST_OUTPUT=$("${PATH_BIN}" "${PATH_CONFIG}/${t}.yaml" "${PATH_CONFIG}/${t}.golden.json" "--details")
 done
 
+# Testing expected matches using --useproto
+# --useproto needs the test schema as a validation.proto message.
+for t in "${TESTS[@]}"
+do
+  TEST_OUTPUT=$("${PATH_BIN}" "${PATH_CONFIG}/${t}.yaml" "${PATH_CONFIG}/${t}.golden.proto.json" "--details" "--useproto")
+done
+
+# Test the yaml test file support
```

Comment:
@hennna Thanks for the idea. I added some tests to demonstrate how to use yaml or pb_text as test files. However, I don't think we should duplicate all tests again. This should be enough to ensure these formats work. WDYT?

Reply:
Agreed. Don't think we need to duplicate all tests again.